### PR TITLE
[0.5] Backporting corrections to GOES satellites operational time ranges

### PR DESCRIPTION
**Pull request by DanRyanIrish**
Hi all,
Here is the backport of the correction I made to the GOES satellite operational time ranges.

### CODE REVIEW

**Comment by Cadair** [comment]
@DanRyanIrish changelog please!

**Comment by DanRyanIrish** [comment]
@Cadair: Done.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ Bug Fixes:
  * Updated SWPC/NOAA links due to their new website.
  * Exposed the raw AIA color tables in `sunpy.cm.color_tables`.
  * Fixes `map` compatibility with Astropy 1.0.x.
+ * Corrected GOES satellites' operational time ranges so all data back to 1980 can be read into GOESLightCurve object
```
Comment (Cadair) [inline CHANGELOG.md]: @DanRyanIrish sorry to be a pain, but seeing how I released 0.5.4 I should have created a 0.5.5 section above. Could you make a new heading and put this under it? Thanks.

Comment (DanRyanIrish) [inline CHANGELOG.md]: Sure.  So just to clarify, the rule is, every PR has a CHANGELOG entry?

Comment (Cadair) [inline CHANGELOG.md]: yes every pr should have a changelog (unless it is a vanishingly stupid change).

Either pypi or github releases (https://github.com/sunpy/sunpy/releases) show the latest version. I should create the new header upon release.

Comment (DanRyanIrish) [inline CHANGELOG.md]: Cool!

**Comment by ehsteve** [comment]
@DanRyanIrish have you actually tested this. After further thought I seem to remember that some of the older files were slightly broken or required further tweaks for reading them.

**Comment by DanRyanIrish** [comment]
@ehsteve: I have tested it, but not every file yet.  I am now in the process of testing every file between 1980 and 2015 just to see if an error is raised when you try to create a GOESLightCurve object.  It's taking a little time.  I'm on 1983 at the moment.  So far I have found a very small number of files that raise a KeyError.  I don't fully understand this yet but it is only a small number of files.

There is a separate issue with pre-GOES-8 fluxes.  The archived values have been incorrectly scaled.  To get true fluxes the short channel must be divided by 0.85 and the long channel by 0.7.  This is recommended by NOAA.  If this isn't already built into the GOESLightCurve object then we should add it.

**Comment by Cadair** [comment]
@DanRyanIrish any changes you make to the GOES downloading code are going to have to be ported to the unified downloader branch. Once it's done we can worry about that tho.

**Comment by DanRyanIrish** [comment]
@Cadair I don't believe at the moment that the issue is in downloading the files, but the files themselves.  So I'm not planning on changing the GOES downloading code at this time.

**Comment by dpshelio** [comment]
Just for the records, this is a backport of #1319 

:+1:  - but a confirmation from the big check is appreciated :)
Also, @DanRyanIrish could you make a list of the files that fails and make a issue with them and link it with this or the other PR for future reference if someone finds them again.

**Comment by DanRyanIrish** [comment]
@dpshelio Sure, I can do that.

**Comment by DanRyanIrish** [comment]
For a list of dates in the period 1980-2014 for which the GOESLightCurve crashed when you try to create it, see Issue #1337.

**Pull request closed without merging**